ideascube/library/tests/test_views.py

```diff
@@ -108,6 +114,26 @@ def test_book_detail_page_does_not_list_specimens_to_non_staff(app, book):
     resp.mustcontain(no=[specimen.barcode])
 
 
+def test_book_detail_page_shows_warning_message_to_staff_if_no_specimens(staffapp, book):
+    resp = staffapp.get(reverse('library:book_detail', kwargs={'pk': book.pk}))
+    resp.mustcontain("Please add a specimen, or the book won&#39;t be available for the users")
+
+
+def test_book_detail_page_does_not_show_warning_message_to_user_if_no_specimens(app, book):
```

Comment:
So users can reach the book page, even though they can't see the book in the list?

Is that what we want? Or should the detail page for a book without specimens instead be a 404 if the user isn't staff?